### ISSUE TITLE
[Challenge GradeBook]  two functions to the same 

### ISSUE THREAD

**Issue by rwobben** [exercism/xcsharp#271]
Hello 

I noticed that the grade and Roster have the same type definition and so do the same. 
If I put the code of Roster into Grade all tests will be green. 

Regards, 

Roelof

**Comment by ErikSchierboom** [comment]
Good catch! We'll be fixing this.

**Comment by rwobben** [comment]
Thanks, if I can help , ask

**Comment by ErikSchierboom** [comment]
@rwobben Well, a PR would be appreciated! What I would like to see is two things:

1. Move the instance field (`private readonly School school = new School();`) to be a variable in each test (makes it easier to read).
1. The `Roster` function should not take any arguments, and should return the names of all student, sorted alphabetically. That means that the final test ("Student_names_in_each_grade_in_roster_are_sorted") should be using the `Roster` function and verifying that the return string enumerable is a sorted enumerable of *all* student names.

You can find the test's source in exercises/grade-school/GradeSchoolTest.cs. Having modified the test, you should then update the default and example implementation, which are in the same folder as the test.

**Comment by rwobben** [comment]
Oke, First one is easy. 
The second can take a little more time. IM just a beginner/profient in c#. 

What do you mean with the default and example implementation ?

**Comment by ErikSchierboom** [comment]
@rwobben If you look in the `exercises/grade-school` directory, you'll find the following two files:

1. `Example.cs`: this file contains a working implementation that passes all tests.
1. `GradeSchool.cs`: this file contains a default implementation, with only the minimum of methods to help the user get started. It should not contain any implementation.

Would you mind creating to separate PR's? So first the one where the instance field is moved, and then once that has merged, the bigger `Roster` change? It is always a good idea to create a PR that is as small as possible, it helps us review the code and you have a better chance of getting the changes merged sooner.

**Comment by rwobben** [comment]
No problem.  So also make multiple issues.
The only things is that I use LINQ a lot to solve the exercises. So my solutions looks very different from yours.  I do not have the class SortedList in my code because it's not needed for solving it

**Comment by rwobben** [comment]
One question. 

Is it allright with you I change the test to different parts 
So a arrange where all things are set up
A act part where the actual calling is done
A Assert part where the actual checking take place 

I think the test are then easier to read

**Comment by ErikSchierboom** [comment]
@rwobben  Usually, I think the AAA pattern is fine, but we don't really use it a lot here. Also, there is no value returned by `Add`, so I would suggest the following:

```csharp
[Fact]
public void Adding_a_student_adds_them_to_the_roster_for_the_given_grade()
{
    var school = new School();
    school.Add("Aimee", 2);

    var actual = school.Roster(2); 

    var expected = new[] { "Aimee" };
    Assert.Equal(expected, actual);
}
```

**Comment by rwobben** [comment]
Oke, then the first test is rewritten

**Comment by ErikSchierboom** [comment]
@rwobben Okay, looking forward to the PR.

**Comment by ErikSchierboom** [comment]
@rwobben I've requested some changes to the PR. You should have received a notification.

**Comment by ErikSchierboom** [comment]
Closed by #273.
